### PR TITLE
Docs/629/animation

### PR DESCRIPTION
**Pull request by azriel91**
Added documentation for animation crate (#629).

There were some bits that didn't seem to fit in the crate documentation, so I put it under `book`, but it feels "halfway" since there isn't a complete tutorial.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/631)

### CODE REVIEW

File: amethyst_animation/src/resources.rs [hunk 1]
```diff
@@ -42,10 +45,25 @@ where
     T: InterpolationPrimitive,
 {
     /// Time of key frames
+    ///
+    /// A simple example of this for animations that are defined with 4 evenly spaced key frames is
+    /// `vec![0., 1., 2., 3.]`.
     pub input: Vec<f32>,
     /// Actual output data to interpolate
+    ///
+    /// For `input` of size `i`, the `output` is size `n * i`, where `n` is the number of values for
```
Comment (Rhuagh) [inline amethyst_animation/src/resources.rs]: This is not completely correct, the size of the output can be anything the function requires, so there's no specific limit on this size.

Comment (Rhuagh) [reply amethyst_animation/src/resources.rs]: A summary of the predefined functions would be:

* Linear - i
* SphericalLinear - i
* CubicSpline - i * 3
* CatmullRomSpline - i + 2
* Step - i

File: amethyst_animation/src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,50 @@
...
+//! `(Channel, Sampler)` pair is enough to define one part the animation, and a
+//! `Vec<(Channel, Sampler)>` defines the whole animation.
+//!
+//! In a more complex situation, an object in game may be made up of multiple entities. Say you have
+//! a dragon monster, that is defined by a skinned mesh that has a skeleton with 10 joints. Each
+//! joint will then be an `Entity`. Our animation definition holds the samplers to run for the whole
+//! object. To animate each of the entities of this complex object, we need a way to link the
+//! sampler to the each of the entities.
+//!
+//! Animation definitions are persistent and can be stored on disk. Entities however, are not. To
```
Comment (jojolepro) [inline amethyst_animation/src/lib.rs]: Entities are not, but they will soon

File: book/src/animation/definition.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Definition
+
+Animations can be defined for objects made of a single entity, or complex objects made up of multiple entities.
+
+Right now we do not have a tutorial for defining an animation from scratch, but take a look at the following resources to get an idea of how to write one:
```
Comment (jojolepro) [inline book/src/animation/definition.md]: We don't have a tutorial for animations. Do you mind opening an issue for that, and reference this line so we don't forget to change it when we do?

File: amethyst_animation/src/lib.rs (hunk 2 above)
Comment (Rhuagh) [reply amethyst_animation/src/lib.rs]: No, the Entity itself cannot be stored. Markers can.

File: book/src/animation/definition.md (hunk 3 above)
Comment (azriel91) [reply book/src/animation/definition.md]: Certainly :slightly_smiling_face: #637

**Pull request merged**